mat/diagonal.go

```diff
@@ -324,3 +326,24 @@ func (d *DiagDense) Trace() float64 {
 	return tr
 
 }
+
+// Norm returns the specified norm of the receiver. Valid norms are:
+//  1 or Inf - The maximum diagonal element magnitude
+//  2 - The Frobenius norm, the square root of the sum of the squares of
+//      the diagonal elements
+//
+// Norm will panic with ErrNormOrder if an illegal norm is specified.
```

Comment:
" and with ErrShape if the receiver has zero size"

Reply:
Done.